packages/protect/src/lib/protect.ts

```diff
@@ -0,0 +1,261 @@
+      return data._type === 'PingOneProtect';
+    });
+
+    if (!protectMetadataCb) {
+      return 'none';
+    }
+
+    const data = (protectMetadataCb as MetadataCallback).getData() as ProtectInitializeConfig;
+
+    return data._action === 'protect_initialize' ? 'initialize' : 'evaluate';
+  }
+
+  function setNodeClientError(step: FRStep, value: string): void {
+    // Check for native callback first
+    const nativeEvaluationCallback = step.getCallbacksOfType<PingOneProtectEvaluationCallback>(
+      CallbackType.PingOneProtectEvaluationCallback,
+    );
+    const nativeInitializeCallback = step.getCallbacksOfType<PingOneProtectInitializeCallback>(
+      CallbackType.PingOneProtectInitializeCallback,
+    );
+    const arr = [...nativeEvaluationCallback, ...nativeInitializeCallback];
+
+    if (arr.length) {
+      const cb = arr[0];
+      cb.setClientError(value);
+      return;
+    }
+
+    // If we are here, we are dealing with Marketplace nodes
+    const cbs = step.getCallbacksOfType<HiddenValueCallback>(CallbackType.HiddenValueCallback);
+
+    if (!cbs.length) {
+      return;
+    }
+
+    const clientErrorCb = cbs.find((hiddenValueCallback) => {
+      const output = hiddenValueCallback.getOutputByName<string>('id', '');
+      return output === 'clientError';
+    });
+
+    if (!clientErrorCb) {
+      return;
+    }
+
+    clientErrorCb.setInputValue(value);
+  }
+
+  function setNodeInputValue(step: FRStep, value: string): void {
+    // Check for native callback first
+    try {
+      const nativeCallback = step.getCallbackOfType<PingOneProtectEvaluationCallback>(
+        CallbackType.PingOneProtectEvaluationCallback,
+      );
+
+      nativeCallback?.setData(value);
+      return;
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    } catch (err) {
+      // Do nothing
+    }
+
+    // If we are here, we are dealing with Marketplace nodes
+    const cbs = step.getCallbacksOfType<HiddenValueCallback>(CallbackType.HiddenValueCallback);
+
+    if (!cbs.length) {
+      return;
+    }
+
+    const inputCb = cbs.find((hiddenValueCallback) => {
+      const output = hiddenValueCallback.getOutputByName<string>('id', '');
+      return output === 'pingone_risk_evaluation_signals';
+    });
+
+    if (!inputCb) {
+      return;
+    }
+
+    inputCb.setInputValue(value);
+  }
+
+  const protectAPI: Protect = {
+    start,
+    getData,
+    pauseBehavioralData,
+    resumeBehavioralData,
+    getPauseBehavioralData,
+    getNodeConfig,
+    getProtectType,
+    setNodeClientError,
+    setNodeInputValue,
+  };
+
+  try {
+    /*
+     * Load the Ping Signals SDK
+     * this automatically pollutes the window
+     * there are no exports of this module
+     */
+    await import('./signals-sdk.js' as string);
+    return protectAPI;
+  } catch (err) {
+    console.error('error loading ping signals', err);
+    throw new Error('Failed to load PingOne Signals SDK');
+  }
```

Comment:
Can we move this to the top? If this fails, I'd rather just throw early, rather than late.